vunk-lexer/src/lib.rs

```diff
@@ -80,129 +78,160 @@ impl std::fmt::Display for Token {
+                just("=").map(Token::Op),
+                just(":").map(Token::Op),
+                just(".").map(Token::Op),
+                just(";").map(Token::Op),
+                just("[").map(Token::Op),
+                just("]").map(Token::Op),
+                just("{").map(Token::Op),
+                just("}").map(Token::Op),
+                just("|").map(Token::Op),
+                just("+").map(Token::Op),
+                just("-").then_ignore(just(">").not()).map(Token::Op),
+                just("*").map(Token::Op),
+                just("/").map(Token::Op),
+                just("%").map(Token::Op),
+                just("^").map(Token::Op),
+            )),
+        ))
     };
 
-    let kw_use = just("use").map(|_| Token::Use);
-    let kw_pub = just("pub").map(|_| Token::Pub);
-    let kw_arrow = just("->").map(|_| Token::Arrow);
-    let kw_let = just("let").map(|_| Token::Let);
-    let kw_in = just("in").map(|_| Token::In);
-    let kw_if = just("if").map(|_| Token::If);
-    let kw_then = just("then").map(|_| Token::Then);
-    let kw_else = just("else").map(|_| Token::Else);
-    let kw_true = just("true").map(|_| Token::Bool(true));
-    let kw_false = just("false").map(|_| Token::Bool(false));
-    let kw_where = just("where").map(|_| Token::Where);
-    let kw_match = just("match").map(|_| Token::Match);
-    let kw_when = just("when").map(|_| Token::When);
-    let kw_type = just("type").map(|_| Token::Type);
-    let kw_impl = just("impl").map(|_| Token::Impl);
-    let kw_enum = just("enum").map(|_| Token::Enum);
-    let kw_mod = just("mod").map(|_| Token::Mod);
-    let ident = ident().map(Token::Ident);
-
-    // A single token can be one of the above
-    let token = num
-        .or(str_)
-        .or(kw_use)
-        .or(kw_pub)
-        .or(kw_arrow)
-        .or(kw_let)
-        .or(kw_in)
-        .or(kw_if)
-        .or(kw_then)
-        .or(kw_else)
-        .or(kw_true)
-        .or(kw_false)
-        .or(kw_where)
-        .or(kw_match)
-        .or(kw_when)
-        .or(kw_impl)
-        .or(kw_type)
-        .or(kw_enum)
-        .or(kw_mod)
-        .or(ctrl)
-        .or(operator)
-        .or(ident)
-        .recover_with(skip_then_retry_until([]));
+    let ident = ident()
+        .map(|ident: &str| match ident {
+            "if" => Token::If,
+            "then" => Token::Then,
+            "else" => Token::Else,
+            "let" => Token::Let,
+            "in" => Token::In,
+            "where" => Token::Where,
+            "match" => Token::Match,
+            "when" => Token::When,
+            "type" => Token::Type,
+            "impl" => Token::Impl,
+            "enum" => Token::Enum,
+            "use" => Token::Use,
+            "pub" => Token::Pub,
+            "mod" => Token::Mod,
+            "true" => Token::Bool(true),
+            "false" => Token::Bool(false),
+
+            _ => Token::Ident(ident),
+        })
+        .or(just("$").map(|_| Token::Dollar));
```

Comment:
You can use `.to(Token::Dollar)` instead of `.map(|_| Token::Dollar)`, if you wish.